### DISCUSSION TITLE
Filter and order by content fields

### DISCUSSION THREAD

**Discussion by aghili371** [OrchardCMS/OrchardCore#8221] [Q&A]
I had a question regarding working with queries.

i have ContentType  "Payment" and I added an extra text field  "PaymentType".
how can I filter and sort my list by "PaymentType" filed?

```
    var query = _session.Query<ContentItem, ContentItemIndex>();
            query = query.With<ContentItemIndex>(x => x.Latest);
            query = query.With<ContentItemIndex>(x => x.ContentType == "Payment");
            query = query.With<ContentItemIndex>(x => ?????);
            query = queryAcademyTerm.OrderByDescending(???);

    var pageOfContentItems = await query.ListAsync();
```

**Comment by lampersky** [comment]
you should query for ContentItems combining two indices:
- **ContentItemIndex** - here you can select ContentType in which you are interested in (also you can choose **Latest** and **Published** items only),
- **TextFieldIndex** - using this index, you can filter those Content Items which have Field called **PaymentType**
 
 example:
 
![image](https://user-images.githubusercontent.com/6403130/107151005-b5096200-6960-11eb-8843-eb914a074d51.png)

**Reply by Skrypt** [reply]
Also alternatively consider using Lucene Queries which can be used to achieve the same result.
